src/wallet/wallet.h

```diff
@@ -66,6 +75,16 @@ static const CAmount MIN_CHANGE = CENT;
 static const bool DEFAULT_SPEND_ZEROCONF_CHANGE = true;
 //! Default for -sendfreetransactions
 static const bool DEFAULT_SEND_FREE_TRANSACTIONS = false;
+//! Defaults for -allowdeprecated
+static const std::set<std::string> DEFAULT_ALLOW_DEPRECATED{{
+    "legacy_privacy",
+    "getnewaddress",
+    "z_getnewaddress",
+    "zcrawreceive",
+    "zcrawjoinsplit",
+    "zcrawkeygen",
+    "addrtype"
+}};
```

Comment:
In a pairing with @nuttycom, we decided on a more unified approach:

- Have two lists in `src/deprecation.cpp`:
  - A list of all deprecated features that are default-allowed.
  - A list of all deprecated features that are default-denied.
- Add helper functions to:
  - Check that all provided flags refer to a known deprecated feature (called on init).
    - It is a user error to try to enable a feature we've removed, or that we haven't deprecated yet. This is important to convey because it's a chance to point out to the user that, no really, this feature is gone now.
  - Check whether a specific feature is enabled.
    - Checks the flags (specific or `all/none`), then the defaults.
- Document a feature deprecation policy:
  - When we deprecate a feature, we add a flag for it (to enable users to test their environments without it), but leave it default-allowed.
  - After X releases / time-period, we move it to default-denied (but users can still re-enable it if necessary).
  - After Y more releases / time-period, we remove the feature and its flag.